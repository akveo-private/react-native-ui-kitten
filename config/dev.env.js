const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
