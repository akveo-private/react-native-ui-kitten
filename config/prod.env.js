const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
