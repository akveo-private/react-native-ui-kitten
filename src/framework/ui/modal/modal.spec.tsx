import React from 'react';
import {
  fireEvent,
  render,
} from 'react-native-testing-library';
import {
  View,
  Text,
  TouchableWithoutFeedback,
  Button,
} from 'react-native';
import {
  Modal,
  Props as ModalProps,
  ModalAnimationType,
} from './modal.component';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import * as config from './modal.spec.config';

jest.useFakeTimers();

const StyledModal = styled<Modal, ModalProps>(Modal);

const Mock = (props?: ModalProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <StyledModal {...props} />
  </StyleProvider>
);

describe('@modal component checks', () => {

  const MODAL_TEST_IDENTIFIER: (substring: string) => string = (substring: string): string =>
    `modal-test-identifier-${substring}`;

  interface TestScreenProps {
    modalAnimationType: ModalAnimationType;
    buttonTestId: string;
  }

  interface TestScreenState {
    modalVisible: boolean;
  }

  class TestScreen extends React.Component<TestScreenProps, TestScreenState> {

    public state: TestScreenState = {
      modalVisible: false,
    };

    private setModalVisible(value: boolean): void {
      this.setState({ modalVisible: value });
    }

    public render(): React.ReactNode {
      return (
        <View>
          <Button testID={this.props.buttonTestId}
                  title={'Open Modal'}
                  onPress={() => this.setModalVisible(true)}/>
          <Modal
            visible={this.state.modalVisible}
            isBackDropAllowed={true}
            identifier={MODAL_TEST_IDENTIFIER('1')}
            onCloseModal={() => 1}
            animationType={this.props.modalAnimationType}
            animationDuration={500}
          >
            <View><Text>Test2</Text></View>
          </Modal>,
        </View>
      );
    }
  }

  it('* modal component renders properly', () => {
    const modal1 = render(
      <Modal
        visible={true}
        isBackDropAllowed={false}
        identifier={MODAL_TEST_IDENTIFIER('1')}
        onCloseModal={() => 1}
      >
        <View><Text>Test1</Text></View>
      </Modal>,
    );
    const modal2 = render(
      <Modal
        visible={false}
        isBackDropAllowed={false}
        identifier={MODAL_TEST_IDENTIFIER('1')}
        onCloseModal={() => 1}>
        <View><Text>Test2</Text></View>
      </Modal>,
    );

    expect(modal1).toMatchSnapshot();
    expect(modal2).toMatchSnapshot();
  });

  it('* with animations', () => {
    const testApplication1 = render(<TestScreen modalAnimationType='slideInUp' buttonTestId='1'/>);
    const testApplication2 = render(<TestScreen modalAnimationType='fade' buttonTestId='2'/>);
    fireEvent.press(testApplication1.getByTestId('1'));
    fireEvent.press(testApplication2.getByTestId('2'));

    expect(testApplication1).toMatchSnapshot();
    expect(testApplication2).toMatchSnapshot();
  });

  it('* modal component props checks', () => {
    const modalPassingProps = {
      visible: true,
      isBackDropAllowed: false,
    };
    const modal = <Modal {...modalPassingProps}/>;

    expect(modal.props.visible).toBe(modalPassingProps.visible);
    expect(modal.props.isBackDropAllowed).toBe(modalPassingProps.isBackDropAllowed);
  });

  it('* modal closes on passed prop', () => {
    const onCloseModal = jest.fn();
    const component =
      <Modal
        visible={true}
        isBackDropAllowed={true}
        identifier={MODAL_TEST_IDENTIFIER('1')}
        onCloseModal={onCloseModal}
      >
        <View>
          <Text>Test1</Text>
          <Button title={'Close Modal'} onPress={onCloseModal}/>
        </View>
      </Modal>;
    const modal = render(component);
    expect(modal).toMatchSnapshot();
    fireEvent.press(modal.getByType(Button));
    expect(onCloseModal).toHaveBeenCalled();
    expect(modal).toMatchSnapshot();
  });

  it('* modal component close on backDrop checks', () => {
    const onCloseModal = jest.fn();
    const modal = render(
      <Modal
        visible={true}
        isBackDropAllowed={true}
        identifier={MODAL_TEST_IDENTIFIER('1')}
        onCloseModal={onCloseModal}>
        <View><Text>Test1</Text></View>
      </Modal>,
    );
    expect(modal).toMatchSnapshot();
    fireEvent.press(modal.getByType(TouchableWithoutFeedback));
    expect(modal).toMatchSnapshot();
    expect(onCloseModal).toHaveBeenCalled();
  });

  it('* component styled with mappings', () => {
    const component = render(
      <Mock visible={true}>
        <Text>Test</Text>
      </Mock>,
    );
    expect(component).toMatchSnapshot();
  });

});
