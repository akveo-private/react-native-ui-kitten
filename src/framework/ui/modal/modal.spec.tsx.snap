// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modal component checks * component styled with mappings 1`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    animationDuration={300}
    appearance="default"
    dispatch={[Function]}
    isBackDropAllowed={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#3366FF",
        "borderRadius": 3,
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
    theme={
      Object {
        "blue-dark": "#2541CC",
        "blue-primary": "#3366FF",
        "gray-dark": "#8992A3",
        "gray-highlight": "#EDF0F5",
        "gray-light": "#DDE1EB",
        "gray-primary": "#A6AEBD",
        "pink-primary": "#FF3D71",
      }
    }
    visible={true}
  >
    <Text
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`@modal component checks * modal closes on passed prop 1`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    animationDuration={300}
    identifier="modal-test-identifier-1"
    isBackDropAllowed={true}
    onCloseModal={[MockFunction]}
    style={Object {}}
    visible={true}
  >
    <View
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      <Text>
        Test1
      </Text>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close Modal
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`@modal component checks * modal closes on passed prop 2`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    animationDuration={300}
    identifier="modal-test-identifier-1"
    isBackDropAllowed={true}
    onCloseModal={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={Object {}}
    visible={true}
  >
    <View
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      <Text>
        Test1
      </Text>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close Modal
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`@modal component checks * modal component close on backDrop checks 1`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    animationDuration={300}
    identifier="modal-test-identifier-1"
    isBackDropAllowed={true}
    onCloseModal={[MockFunction]}
    style={Object {}}
    visible={true}
  >
    <View
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      <Text>
        Test1
      </Text>
    </View>
  </View>
</View>
`;

exports[`@modal component checks * modal component close on backDrop checks 2`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    animationDuration={300}
    identifier="modal-test-identifier-1"
    isBackDropAllowed={true}
    onCloseModal={
      [MockFunction] {
        "calls": Array [
          Array [
            "modal-test-identifier-1",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={Object {}}
    visible={true}
  >
    <View
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      <Text>
        Test1
      </Text>
    </View>
  </View>
</View>
`;

exports[`@modal component checks * modal component renders properly 1`] = `
<View
  style={
    Object {
      "position": "absolute",
    }
  }
>
  <View
    animationDuration={300}
    identifier="modal-test-identifier-1"
    isBackDropAllowed={false}
    onCloseModal={[Function]}
    style={Object {}}
    visible={true}
  >
    <View
      onCloseModal={[Function]}
      pointerEvents="box-none"
    >
      <Text>
        Test1
      </Text>
    </View>
  </View>
</View>
`;

exports[`@modal component checks * modal component renders properly 2`] = `null`;

exports[`@modal component checks * with animations 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="1"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open Modal
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      animationDuration={500}
      identifier="modal-test-identifier-1"
      isBackDropAllowed={true}
      onCloseModal={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      visible={true}
    >
      <View
        onCloseModal={[Function]}
        pointerEvents="box-none"
      >
        <Text>
          Test2
        </Text>
      </View>
    </View>
  </View>
  ,
</View>
`;

exports[`@modal component checks * with animations 2`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="2"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open Modal
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      animationDuration={500}
      identifier="modal-test-identifier-1"
      isBackDropAllowed={true}
      onCloseModal={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
      visible={true}
    >
      <View
        onCloseModal={[Function]}
        pointerEvents="box-none"
      >
        <Text>
          Test2
        </Text>
      </View>
    </View>
  </View>
  ,
</View>
`;
